Netty based inputs don't use tcnative TLS anymore

If Graylog is used together with the enterprise-integrations plugin,
it will fail to use the native openssl TLS support for Netty.

The default `transport_netty_tls_provider = auto` will first try to use openssl and
then fall back to using the jdk TLS.

This is due to the fact that Netty will skip loading the native libraries,
once it detects multiple versions:

https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java#L301

The PR https://github.com/Graylog2/graylog-plugin-enterprise-integrations/pull/662

introduced a transitive dependency through `azure-messaging-eventhubs`.
This leads to having netty-tcnative in version 2.0.40 and version 2.0.54.

## Steps to Reproduce (for bugs)

run graylog with `transport_netty_tls_provider = openssl`
and connect to an TLS enabled input.
This will yield:

```
2023-02-03 15:22:37,972 WARN : io.netty.channel.ChannelInitializer - Failed to initialize a channel. Closing: [id: 0xe9f9f682, L:/127.0.0.1:6044 - R:/127.0.0.1:59312]
java.lang.UnsatisfiedLinkError: failed to load the required native library
	at io.netty.handler.ssl.OpenSsl.ensureAvailability(OpenSsl.java:601) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:215) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.OpenSslContext.<init>(OpenSslContext.java:45) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:349) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:336) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.SslContext.newServerContextInternal(SslContext.java:474) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:606) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at org.graylog2.plugin.inputs.transports.AbstractTcpTransport$3.createSslEngine(AbstractTcpTransport.java:376) ~[classes/:?]
	at org.graylog2.plugin.inputs.transports.AbstractTcpTransport$3.call(AbstractTcpTransport.java:337) ~[classes/:?]
	at org.graylog2.plugin.inputs.transports.AbstractTcpTransport$3.call(AbstractTcpTransport.java:333) ~[classes/:?]
	at org.graylog2.plugin.inputs.transports.NettyTransport$1.initChannel(NettyTransport.java:105) ~[classes/:?]
	at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:129) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:112) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:1114) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.DefaultChannelPipeline.access$100(DefaultChannelPipeline.java:46) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1463) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1115) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:650) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:514) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:167) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569) [netty-transport-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.84.Final.jar:4.1.84.Final]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [metrics-core-4.1.9.jar:4.1.9]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) [metrics-core-4.1.9.jar:4.1.9]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.IllegalArgumentException: Failed to load any of the given libraries: [netty_tcnative_linux_x86_64, netty_tcnative_linux_x86_64_fedora, netty_tcnative_x86_64, netty_tcnative]
	at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:114) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.OpenSsl.loadTcNative(OpenSsl.java:710) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:149) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
	at org.graylog2.plugin.inputs.transports.AbstractTcpTransport.getSecureCipherSuites(AbstractTcpTransport.java:382) ~[classes/:?]
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183) ~[guava-31.0.1-jre.jar:?]
	at org.graylog2.plugin.inputs.transports.AbstractTcpTransport$3.createSslEngine(AbstractTcpTransport.java:371) ~[classes/:?]
	... 26 more
	Suppressed: java.lang.IllegalStateException: Multiple resources found for 'META-INF/native/libnetty_tcnative_linux_x86_64.so' with different content: [jar:file:/home/marco/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.54.Final/netty-tcnative-boringssl-static-2.0.54.Final-linux-x86_64.jar!/META-INF/native/libnetty_tcnative_linux_x86_64.so, jar:file:/home/marco/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.40.Final/netty-tcnative-boringssl-static-2.0.40.Final.jar!/META-INF/native/libnetty_tcnative_linux_x86_64.so]
		at io.netty.util.internal.NativeLibraryLoader.getResource(NativeLibraryLoader.java:301) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:173) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:105) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
		at io.netty.handler.ssl.OpenSsl.loadTcNative(OpenSsl.java:710) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
		at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:149) ~[netty-handler-4.1.84.Final.jar:4.1.84.Final]
		at org.graylog2.plugin.inputs.transports.AbstractTcpTransport.getSecureCipherSuites(AbstractTcpTransport.java:382) ~[classes/:?]
```


* Graylog Version: 5.1.0-SNAPSHOT

--- a/graylog-project-parent/pom.xml
+++ b/graylog-project-parent/pom.xml
@@ -67,25 +67,6 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
-            <dependency>
-                <groupId>io.netty</groupId>
-                <artifactId>netty-bom</artifactId>
-                <version>${netty.version}</version>
-                <type>pom</type>
-                <scope>import</scope>
-            </dependency>
-            <dependency>
-                <groupId>io.netty</groupId>
-                <artifactId>netty-tcnative-boringssl-static</artifactId>
-                <version>${netty-tcnative-boringssl-static.version}</version>
-                <classifier>osx-x86_64</classifier>
-            </dependency>
-            <dependency>
-                <groupId>io.netty</groupId>
-                <artifactId>netty-tcnative-boringssl-static</artifactId>
-                <version>${netty-tcnative-boringssl-static.version}</version>
-                <classifier>linux-x86_64</classifier>
-            </dependency>
 
             <dependency>
                 <groupId>com.google.guava</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -251,6 +251,26 @@
                 <artifactId>httpcore-nio</artifactId>
                 <version>${apache-httpcore.version}</version>
             </dependency>
+            <!-- Ensure consistent Netty versions across dependencies and transitive dependencies -->
+            <dependency>
+                <groupId>io.netty</groupId>
+                <artifactId>netty-bom</artifactId>
+                <version>${netty.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+            <dependency>
+                <groupId>io.netty</groupId>
+                <artifactId>netty-tcnative-boringssl-static</artifactId>
+                <version>${netty-tcnative-boringssl-static.version}</version>
+                <classifier>osx-x86_64</classifier>
+            </dependency>
+            <dependency>
+                <groupId>io.netty</groupId>
+                <artifactId>netty-tcnative-boringssl-static</artifactId>
+                <version>${netty-tcnative-boringssl-static.version}</version>
+                <classifier>linux-x86_64</classifier>
+            </dependency>
         </dependencies>
     </dependencyManagement>
 

